### PR TITLE
Katalyst revamp permission3 test

### PR DESCRIPTION
**Pull request by ducquangkstn**

### CODE REVIEW

File: test/sol5/withdrawable3.js [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+let withdrawableInst;
+let initialTokenBalance = new BN(100);
+let tokenWithdrawAmt = new BN(60);
+let initialEtherBalance = new BN(10);
+let etherWithdrawAmt = new BN(3);
+
+const {zeroBN} = require("../helper.js");
+const { expectRevert } = require('@openzeppelin/test-helpers');
+
+contract('Withdrawable2', function(accounts) {
```
Comment (ilanDoron) [inline test/sol5/withdrawable3.js:19]: wrong name

File: contractsSol5/KyberStorage.sol [hunk 2]
```diff
@@ -151,7 +160,7 @@ contract KyberStorage is IKyberStorage {
         onlyNetwork
         returns (bool)
     {
-        require(kyberProxyArray.length < max_approved_proxies, "Max 2 proxy");
+        require(kyberProxyArray.length < max_approved_proxies, "Max proxy");
```
Comment (ilanDoron) [inline contractsSol5/KyberStorage.sol:163]: max proxies

File: contractsSol5/mock/MockPermissionGroups3.sol [hunk 3]
```diff
@@ -0,0 +1,26 @@
+pragma solidity 0.5.11;
+
+import "../utils/PermissionGroups3.sol";
+
+
+contract MockPermission2 is PermissionGroups3 {
```
Comment (ilanDoron) [inline contractsSol5/mock/MockPermissionGroups3.sol:6]: have contract name same as file name.
easier later

File: test/sol5/kyberNetwork.js [hunk 4]
```diff
@@ -128,7 +128,7 @@ contract('KyberNetwork', function(accounts) {
         platformFeeBps = new BN(0);
     });
 
-    describe("should test adding contracts, and adding / removing proxy.", async() => {
+    describe.only("should test adding contracts, and adding / removing proxy.", async() => {
```
Comment (ilanDoron) [inline test/sol5/kyberNetwork.js:131]: should remove only...

File: test/sol5/kyberStorage.js [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+            });
+
+            it("should be able to re-add a reserve after its removal", async() => {
+                await tempStorage.removeReserve(reserve.address, new BN(0), {from: network});
+                await tempStorage.addReserve(reserve.address, reserve.reserveId, {from: network});
+            });
+        });
+    });
+    
+});
```
Comment (ilanDoron) [inline test/sol5/kyberStorage.js:114]: line missing

File: test/sol5/permissionGroups3.js [hunk 6]
```diff
@@ -0,0 +1,358 @@
...
+
+let mainAdmin;
+let secondAdmin;
+let operator;
+let alerter;
+let user;
+
+
+contract('PermissionGroups2', function(accounts) {
+    before("init global accounts", async() => {
```
Comment (ilanDoron) [inline test/sol5/permissionGroups3.js:20]: change contract name.

File: test/sol5/networkHelper.js [hunk 7]
```diff
@@ -308,7 +309,7 @@ module.exports.removeReservesFromNetwork = async function (networkInstance, rese
 
 module.exports.genReserveID = genReserveID; 
 function genReserveID(reserveID, reserveAddress) {
-    return reserveID + reserveAddress.substring(2,20);
+    return reserveID + reserveAddress.substring(2,20) + "0".repeat(38);
```
Comment (ilanDoron) [inline test/sol5/networkHelper.js:312]: this fails proxy tests.
not sure why
reverted this change.

File: contractsSol5/KyberStorage.sol [hunk 8]
```diff
@@ -227,11 +255,19 @@ contract KyberStorage is IKyberStorage, PermissionGroups2 {
         }
     }
 
-    function getReservesPerTokenSrc(address token) external view returns (bytes32[] memory reserveIds) {
+    function getReservesPerTokenSrc(address token)
```
Comment (ilanDoron) [inline contractsSol5/KyberStorage.sol:258]: this isn't coding style we use in other files.
I am Ok with this style
just have to be aligned.

which style is it based on?

File: contractsSol5/KyberStorage.sol [hunk 9]
```diff
@@ -198,18 +217,27 @@ contract KyberStorage is IKyberStorage, PermissionGroups2 {
         return reserves;
     }
 
-    function convertReserveAddresstoId(address reserve) external view returns (bytes32 reserveId) {
+    function convertReserveAddresstoId(address reserve)
```
Comment (ilanDoron) [inline contractsSol5/KyberStorage.sol:220]: also here not coding style we use in other files.

File: contractsSol5/KyberStorage.sol [hunk 10]
```diff
@@ -62,7 +66,6 @@ contract KyberStorage is IKyberStorage, PermissionGroups2 {
         } else {
             matchingEngine.push(newMatchingEngine);
         }
-
```
Comment (ilanDoron) [inline contractsSol5/KyberStorage.sol:65]: usually if statement should be separated with empty line

File: contractsSol5/KyberStorage.sol [hunk 11]
```diff
@@ -21,33 +21,37 @@ contract KyberStorage is IKyberStorage, PermissionGroups2 {
...
+    mapping(bytes32 => address[]) public reserveIdToAddresses;
+    mapping(address => bytes32) internal reserveAddressToId;
+    mapping(address => bytes32[]) internal reservesPerTokenSrc; // reserves supporting token to eth
+    mapping(address => bytes32[]) internal reservesPerTokenDest; // reserves support eth to token
 
     IKyberNetwork public network;
 
     modifier onlyNetwork() {
-        require(msg.sender == address(network), "Only network");
+        require(msg.sender == address(network), "ONLY_NETWORK");
```
Comment (ilanDoron) [inline contractsSol5/KyberStorage.sol:32]: please revert this change.

Only network is what we use in other places.

File: contractsSol5/KyberStorage.sol (hunk 8 above)
Comment (ducquangkstn) [reply contractsSol5/KyberStorage.sol:258]: generated by https://github.com/prettier-solidity/prettier-plugin-solidity

File: test/sol5/networkHelper.js (hunk 7 above)
Comment (ducquangkstn) [reply test/sol5/networkHelper.js:312]: After debugging, the problem here is from https://github.com/KyberNetwork/smart-contracts/blob/katalystRevamp/test/sol5/networkHelper.js#L333.
